relay/server.go

```diff
@@ -200,9 +219,22 @@ func (s *Server) GetChunks(ctx context.Context, request *pb.GetChunksRequest) (*
 			"too many chunk requests provided, max is %d", s.config.MaxKeysPerGetChunksRequest)
 	}
 
-	// Future work: client IDs will be fixed when authentication is implemented
-	clientID := fmt.Sprintf("%d", request.RequesterId)
+	client, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, errors.New("could not get peer information")
+	}
+	clientAddress := client.Addr.String()
```

Comment:
should we place this block inside the conditional (L228)?

Reply:
Good idea. Done.